core/eolearn/core/eodata_io.py

```diff
@@ -47,9 +55,9 @@ def save_eopatch(eopatch, filesystem, patch_location, features=..., overwrite_pe
 
     features_to_save = []
     for ftype, fname, path in eopatch_features:
-        feature_io = FeatureIO(filesystem, path)
+        feature_io = FeatureIO(filesystem, path, ftype)
         data = eopatch[(ftype, fname)]
-        file_format = FileFormat.NPY if ftype.is_raster() else FileFormat.PICKLE
+        file_format = _get_file_format(ftype)
```

Comment:
I considered this but I don't want to tie `FileFormat` to `FeatureType`. In the future certain feature types might support multiple file formats. So I would prefer that `FileFormat` and `FeatureType` remain independent from each other.

Reply:
I agree, but even in case that `FeatureType` will support several types (e.g. `FeatureType.DATA` in `npy`, `tiff`, `zarr`...), one will have to be the default, and the code will have to be changed so that the user specifies his preferred file type. In both cases, this will have to be tied to the `feature` itself. 

At the moment, when we only have the default file type, it still makes sense to me to have this tied to the `FeatureType`, as they are not really independent.

Reply:
Hm, I kinda hoped that `FeatureType` would contain only the in-memory logic for features and `FileFormat` only file format logic. That is because nowhere else in eo-learn you need to know about serialization formats of certain feature types.

But if you insist I'll add a method /property about this to `FeatureType` class.

Reply:
They are different architectural approaches, I guess. I don't insist; if you feel more comfortable this way, keep it as is. 

Personally I find the delegation of responsibilities a bit weird.

Reply:
Done in https://github.com/sentinel-hub/eo-learn/pull/348